### DISCUSSION TITLE
Custom file path option for next.config.js file through CLI commands

### DISCUSSION THREAD

**Discussion by amir-zahedi** [vercel/next.js#31433] [Ideas]
### Describe the feature you'd like to request

When abstracting nextJS configs (say in the case of a multi nextJS app mono repo) it can be helpful if one could share a single config file between them. This requires the cli to expose an option for a  custom config file path instead of (as I understand it) only being able to customise nextJS configs through a `next.config.js` file sitting at the app root.

### Describe the solution you'd like

Add support for a -config option or equivalent for cli commands 
next dev -config [CUSTOM CONFIG FILE PATH]
next build -config [CUSTOM CONFIG FILE PATH]
next start -config [CUSTOM CONFIG FILE PATH]

### Describe alternatives you've considered

One possible viable alternative I tried is Importing shared config  into each apps next/config.js that is not the most elegant solution but also in my case failed with `Cannot use import statement outside a module` because all our configs leave in an external npm package.

**Comment by yassernasc** [comment]
I was seeking this `path option` for config files but for another use case besides "merge shared configs". At each config file I would like to use `exportPathMap` to filter pages based on its context. Eg. next.config.app.js, next.config.blog.js, next.config.admin.js.

This would be a very simple "monorepo", reusing components and  package configurations (eslint, tailwind, jest, ..) and just filtering pages at export time.

**Comment by tholder** [comment]
Agree. You should be able to specify next config on command line. We are trying to manage multiple projects from the same code base and each needs a separate config.

**Comment by lake2** [comment]
any updates?

**Comment by xiamu14** [comment]
any updates?

**Comment by wfl-junior** [comment]
+1 I want this as well

**Comment by riosant** [comment]
Did you find any solution?

**Comment by gaojude** [comment]
any updates?

**Comment by arvind-iyer-dev** [comment]
Can we create an Issue for this ?
I want to be able to set the output field to standalone through CLI
maybe the `next build` command can have options which override certain fields in the config file.

Or...

We can set an Env variable like how Bundle Analyzer is enabled ? and modify the values through that

**Comment by coderfin** [comment]
I found a workaround. 

I wanted to be able to easily customize the output of my build.  Initially I was looking for a way to have different config files but then I thought of environment variables.

I setup my `package.json` scripts with an environment variable `OUTPUT`:
```json
...
"scripts": {
  ...
  "build": "next build",
  "export": "OUTPUT=export next build",
  "start": "next start",
  ...
},
...
```

Then in `next.config.mjs`:
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  output: process.env.OUTPUT,
};

export default nextConfig;
```

You could use a similar strategy to have any number of config objects that are used or ignored based on one or more environment variables.

**Reply by coderfin** [reply]
I just realized that @arvind-berrybox had the same or similar suggestion... haha.
https://github.com/vercel/next.js/discussions/31433#discussioncomment-9553184
